connect/json/src/main/java/org/apache/kafka/connect/json/JsonConverter.java

```diff
@@ -724,9 +696,19 @@ private static Object convertToConnect(Schema schema, JsonNode jsonValue) {
                     break;
             }
         }
-        typeConverter = TO_CONNECT_CONVERTERS.get(schemaType);
+
+        final JsonToConnectTypeConverter typeConverter = TO_CONNECT_CONVERTERS.get(schemaType);
         if (typeConverter == null)
-            throw new DataException("Unknown schema type: " + schema.type());
+            throw new DataException("Unknown schema type: " + schemaType);
+
+        if (jsonValue.isNull()) {
+            assert schema != null; // otherwise we would have thrown above because of unknown schema type
```

Comment:
My inline comment is wrong... the reason we can't get here with `schema` null is because of https://github.com/apache/kafka/blob/d7bffebca05aa91a69da03fe29add4f29b964fd9/connect/json/src/main/java/org/apache/kafka/connect/json/JsonConverter.java#L697 and this block being in `jsonValue.isNull()`

But the flow is pretty weird, I'll update. Probably makes sense to move this upto inside the `schema != null` block where `schemaType` is assigned.